services/shhext/chat/encryption.go

```diff
@@ -0,0 +1,468 @@
+package chat
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"errors"
+
+	ecrypto "github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/ecies"
+	"github.com/ethereum/go-ethereum/log"
+	dr "github.com/status-im/doubleratchet"
+
+	"sync"
+	"time"
+
+	"github.com/status-im/status-go/services/shhext/chat/crypto"
+)
+
+var errSessionNotFound = errors.New("session not found")
+
+// EncryptionService defines a service that is responsible for the encryption aspect of the protocol
+type EncryptionService struct {
+	log            log.Logger
+	persistence    PersistenceServiceInterface
+	installationID string
+	mutex          *sync.Mutex
```

Comment:
`mutex sync.Mutex` is fine, no need to make it a pointer. A default value is unlocked.

Reply:
I don't think it is used btw.